maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -349,6 +363,26 @@ private void afterSessionEnd( Collection<MavenProject> projects, MavenSession se
         }
     }
 
+    private void saveResumptionDataWhenApplicable( MavenExecutionResult result, MavenSession session )
```

Comment:
I would remove `WhenApplicable`, it is shorter and the condition doesn't matter (all code is conditional). And why not sync the method name, i.e. `persistResumptionData`

Reply:
I agree, will do.